src/network/core_utils.rs

```diff
@@ -133,45 +140,67 @@ impl CoreUtils {
     async fn add_route_v4(
         handle: &rtnetlink::Handle,
         dest: &Ipv4Network,
-        gateway: &Ipv4Network,
+        gateway: &Ipv4Addr,
     ) -> Result<(), std::io::Error> {
         let route = handle.route();
-        match route
+        let msg = route
             .add()
             .v4()
             .destination_prefix(dest.ip(), dest.prefix())
-            .gateway(gateway.ip())
-            .execute()
-            .await
-        {
-            Ok(_) => Ok(()),
-            Err(err) => Err(std::io::Error::new(
-                std::io::ErrorKind::Other,
-                format!("failed to add route: {}", err),
-            )),
-        }
+            .gateway(*gateway)
+            .message_mut()
+            .to_owned();
+
+        CoreUtils::execute_route_msg(handle, msg).await
```

Comment:
I think we need to propagate the error back upwards

Reply:
It does this, this is just the awkward rust syntax with no semicolon == return